packages/pytest-simcore/src/pytest_simcore/helpers/utils_postgres.py

```diff
@@ -67,3 +65,40 @@ def is_postgres_responsive(url) -> bool:
     except OperationalError:
         return False
     return True
+
+
+async def _insert_and_get_row(
+    conn, table: sa.Table, values: dict[str, Any], pk_col: sa.Column, pk_value: Any
+):
+    result = await conn.execute(table.insert().values(**values).returning(pk_col))
+    row = result.first()
+
+    # NOTE: DO NO USE row[pk_col] since you will get a deprecation error (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
+    assert getattr(row, pk_col.name) == pk_value
+
+    result = await conn.execute(sa.select(table).where(pk_col == pk_value))
+    return result.first()
+
+
+@asynccontextmanager
+async def insert_and_get_row_lifespan(
+    sqlalchemy_async_engine: AsyncEngine,
+    *,
+    table: sa.Table,
+    values: dict[str, Any],
+    pk_col: sa.Column,
+    pk_value: Any,
+) -> AsyncIterator[dict[str, Any]]:
+    # insert & get
+    async with sqlalchemy_async_engine.begin() as conn:
+        row = await _insert_and_get_row(
+            conn, table=table, values=values, pk_col=pk_col, pk_value=pk_value
+        )
+
+    # NOTE: DO NO USE dict(row) since you will get a deprecation error (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
+    # pylint: disable=protected-access
+    yield row._asdict()
```

Comment:
Here's how chatGPT suggests to get a `dict` from a `row`
```python
from sqlalchemy.inspection import inspect

def row_to_dict(row):
    return {c.key: getattr(row, c.key) for c in inspect(row).mapper.column_attrs}
```
Might be worth a try 🤔